### PR TITLE
Fix typings for new Maybe, Maybe.maybe, maybe, new Either, either ...

### PR DESCRIPTION
**Pull request by Ailrun**

### CODE REVIEW

**Comment by cbowdon** [comment]
Thanks for your contribution. I will need to take some time to review this, as I see these are potential breaking changes.

**Comment by Ailrun** [comment]
@cbowdon If users use this library with respecting the semantics of either, this should have no breaking issues.

**Comment by cbowdon** [comment]
You'd hope they do, but I wouldn't assume. :-)

**Comment by Ailrun** [comment]
Ye of course. I'll wait.

**Comment by Ailrun** [comment]
@cbowdon is there any progress?

**Comment by patrickmichalina** [comment]
Any update for this? I am also having issues with the method `maybe()`. The typing is not correct like it is for `Maybe.maybe()`

File: src/maybe.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export type OptionalMaybePatterns<T,U> = Partial<MaybePatterns<T,U>>
  *     or undefined, the Maybe object is filled with Nothing.
  * @see Maybe#
  */
-export function maybe<T>(t: T) {
+export function maybe<T>(t?: T | null | undefined) {
```
Comment (patrickmichalina) [inline src/maybe.ts:44]: Thank you!

**Comment by Ailrun** [comment]
There's no response more than 6 months, so I just gave this up.
Now I try to manage my own FP package, namely [Typed-F](https://github.com/Ailrun/Typed-F).

**Pull request closed without merging**